src/viewProviders/flinkDatabase.ts

```diff
@@ -1,179 +1,373 @@
-    [FlinkDatabaseViewProviderMode.AI, this.aiDelegate],
-  ]);
+    const database: CCloudFlinkDbKafkaCluster = this.database;
+    this.logger.debug(
+      `refreshing entire Flink Database view for ${database.name} (${database.id})`,
+    );
 
-  constructor() {
-    super();
+    await this.withProgress(
+      "Loading Flink Database resources...",
+      async () => {
+        await Promise.allSettled([
+          this.refreshRelationsContainer(database, forceDeepRefresh),
+          this.refreshArtifactsContainer(database, forceDeepRefresh),
+          this.refreshUDFsContainer(database, forceDeepRefresh),
+          this.refreshAIConnectionsContainer(database, forceDeepRefresh),
+          this.refreshAIToolsContainer(database, forceDeepRefresh),
+          this.refreshAIModelsContainer(database, forceDeepRefresh),
+          this.refreshAIAgentsContainer(database, forceDeepRefresh),
+        ]);
+      },
+      false,
+    );
 
-    // Start in relations mode by default.
-    this.defaultDelegate = this.relationsDelegate;
-    this.currentDelegate = this.defaultDelegate;
+    this._onDidChangeTreeData.fire();
   }
 
   setCustomEventListeners(): Disposable[] {
     return [
-      flinkDatabaseViewMode.event(this.switchMode.bind(this)),
       artifactsChanged.event(this.artifactsChangedHandler.bind(this)),
       udfsChanged.event(this.udfsChangedHandler.bind(this)),
     ];
   }
 
+  /** Get resources for the given `container` based on the provided `database` and `loaderMethod`. */
+  private async refreshResourceContainer<T extends FlinkDatabaseResource | FlinkArtifact>(
+    database: CCloudFlinkDbKafkaCluster,
+    container: FlinkDatabaseResourceContainer<FlinkDatabaseResource | FlinkArtifact>,
+    loaderMethod: (database: CCloudFlinkDbKafkaCluster, forceDeepRefresh: boolean) => Promise<T[]>,
+    forceDeepRefresh: boolean = false,
+  ): Promise<T[]> {
+    this.logger.debug(
+      `refreshing ${container.label} resources for ${database.name} (${database.id})...`,
+    );
+    // set initial loading state
+    container.isLoading = true;
+    this._onDidChangeTreeData.fire(container);
+
+    let results: T[] = [];
+    try {
+      results = await loaderMethod(database, forceDeepRefresh);
+      // clear any loading/error state and only refresh the provided container to show updated items
+      container.children = results;
+      container.tooltip = new CustomMarkdownString();
+      container.hasError = false;
+      this._onDidChangeTreeData.fire(container);
+    } catch (error) {
+      let errorMsg = String(error);
+      let errorLanguage = "";
+      // only applies to loading artifacts, since all others are loaded via background statements
+      // and won't throw HTTP response errors
+      if (isResponseError(error)) {
+        const responseBody = await extractResponseBody(error);
+        errorMsg = responseBody?.message || JSON.stringify(responseBody, null, 2);
+        errorLanguage = "json";
+      }
```

Comment:
I opted for simplifying the behavior here and getting rid of the old helper function: https://github.com/confluentinc/vscode/blob/5cc9428e34082e5fe22a9d5a10c6fed6bba21fd2/src/viewProviders/multiViewDelegates/flinkArtifactsDelegate.ts#L34

Since we aren't floating these as error notifications and can instead treat them similar to direct connection errors where only the icon and tooltip change:

<img width="948" height="368" alt="Image" src="https://github.com/user-attachments/assets/fc7e6b40-acc4-4809-a271-74ecdce42703" />